rubix-core/src/main/java/com/qubole/rubix/core/NonLocalReadRequestChain.java

```diff
@@ -133,7 +133,7 @@ public Long call()
             throw e;
           }
           if (nread == -1) {
-            throw new Exception("Error reading from Local Transfer Server");
+            throw new IOException("Error reading from Local Transfer Server");
```

Comment:
nit: Make the exception class changes in separate commit

Reply:
done